### PR TITLE
feat: support change power state to shelved_offloaded

### PR DESCRIPTION
**Pull request by lxShaDoWxl**
support shelve instance

### CODE REVIEW

**Review by ozerovandrei — APPROVED** [review]
LGTM but we need tests to be fixed first https://github.com/terraform-provider-openstack/terraform-provider-openstack/issues/1253

**Comment by nikParasyr** [comment]
recheck

File: openstack/resource_openstack_compute_instance_v2.go [hunk 1]
```diff
@@ -768,6 +768,26 @@ func resourceComputeInstanceV2Update(d *schema.ResourceData, meta interface{}) e
 		powerStateOldRaw, powerStateNewRaw := d.GetChange("power_state")
 		powerStateOld := powerStateOldRaw.(string)
 		powerStateNew := powerStateNewRaw.(string)
+		if strings.ToLower(powerStateNew) == "shelved_offloaded" {
+			err = shelveunshelve.Shelve(computeClient, d.Id()).ExtractErr()
+			if err != nil {
+				return fmt.Errorf("Error shelve OpenStack instance: %s", err)
+			}
+			shelveStateConf := &resource.StateChangeConf{
+				//Pending:    []string{"ACTIVE"},
```
Comment (ozerovandrei) [inline openstack/resource_openstack_compute_instance_v2.go:777]: Is this comment needed?

Comment (ozerovandrei) [reply openstack/resource_openstack_compute_instance_v2.go:777]: I guess it's not critical to have it.

**Review by ozerovandrei — APPROVED** [review]

**Pull request merged**